Assets/Scripts/Audio/AudioController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+
+public class AudioController : MonoBehaviour {
+
+    private float maxVolume;
+	private float fadeTime;
+
+	public bool hasPlayer;
+
+    private BoxCollider2D audioRangeCollider;
+    private AudioSource audioSource;
+    public List<GameObject> dolls = new List<GameObject>();
+
+    // Use this for initialization
+    void Start() {
+        audioRangeCollider = GetComponent<BoxCollider2D>();
+        audioSource = GetComponent<AudioSource>();
+		maxVolume = MechanicAudioManager.getInstance ().maxVolume;
+		fadeTime = MechanicAudioManager.getInstance ().fadeTime;
+    }
+		
+
+	public void updateController(){
+		Debug.Log ("HAHAHA");
+		GameObject player = dolls.Find(x => x.CompareTag("Player"));
+		if (player == null && hasPlayer) {
+			hasPlayer = false;
+			fadeOut ();
+		}
+	}
+
+    //Add the possible player to the list of dolls
+    void OnTriggerEnter2D(Collider2D other) {
+		if (other.gameObject.CompareTag("Player")) {
+			addDoll (other.gameObject);
+        }
+    }
+
+	void OnTriggerStay2D(Collider2D other) {
```

Comment:
I cant exaclt remember the scenario where I need it, I remember it being the case in conveyor belt